flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/StreamExecutionEnvironment.java

```diff
@@ -1096,12 +1096,7 @@ private StateBackend loadStateBackend(ReadableConfig configuration, ClassLoader
         // must not have null elements and mixed elements
         FromElementsFunction.checkCollection(data, typeInfo.getTypeClass());
 
-        SourceFunction<OUT> function;
-        try {
-            function = new FromElementsFunction<>(typeInfo.createSerializer(getConfig()), data);
-        } catch (IOException e) {
-            throw new RuntimeException(e.getMessage(), e);
-        }
+        SourceFunction<OUT> function = new FromElementsFunction<>(data);
```

Comment:
Could we add a single positive test for the `returns` method on `fromElements`?

Reply:
Add `StreamExecutionEnvironmentTest .testFromElementsPostConstructionType`.